utils/syscall.c

```diff
@@ -382,9 +382,342 @@ static const myst_syscall_pair_t _pairs[] = {
+      SYS_semget,
+      SYS_semop,
+      SYS_semctl,
+      SYS_shmdt,
+      SYS_msgget,
+      SYS_msgsnd,
+      SYS_msgrcv,
+      SYS_msgctl,
+      SYS_semtimedop}},
+    {"network", 21, {SYS_sendfile,    SYS_socket,     SYS_connect,
+                     SYS_accept,      SYS_sendto,     SYS_recvfrom,
+                     SYS_sendmsg,     SYS_recvmsg,    SYS_shutdown,
+                     SYS_bind,        SYS_listen,     SYS_getsockname,
+                     SYS_getpeername, SYS_socketpair, SYS_setsockopt,
+                     SYS_getsockopt,  SYS_getpmsg,    SYS_putpmsg,
+                     SYS_accept4,     SYS_recvmmsg,   SYS_sendmmsg}},
+    {"process",
+     14,
+     {SYS_clone,
+      SYS_fork,
+      SYS_vfork,
+      SYS_execve,
+      SYS_exit,
+      SYS_wait4,
+      SYS_kill,
+      SYS_rt_sigqueueinfo,
+      SYS_tkill,
+      SYS_exit_group,
+      SYS_tgkill,
+      SYS_waitid,
+      SYS_rt_tgsigqueueinfo,
+      SYS_execveat}},
+    {"signal",
+     15,
+     {SYS_rt_sigaction,
+      SYS_rt_sigprocmask,
+      SYS_rt_sigreturn,
+      SYS_pause,
+      SYS_kill,
+      SYS_rt_sigpending,
+      //   SYS_t_sigtimedwait,
+      SYS_rt_sigqueueinfo,
+      SYS_rt_sigsuspend,
+      SYS_sigaltstack,
+      SYS_tkill,
+      SYS_tgkill,
+      SYS_signalfd,
+      SYS_signalfd4,
+      SYS_rt_tgsigqueueinfo}},
+    {"memory",
+     24,
+     {SYS_mmap,
+      SYS_mprotect,
+      SYS_munmap,
+      SYS_brk,
+      SYS_mremap,
+      SYS_msync,
+      SYS_mincore,
+      SYS_madvise,
+      SYS_shmat,
+      SYS_shmdt,
+      SYS_mlock,
+      SYS_munlock,
+      SYS_mlockall,
+      SYS_munlockall,
+      SYS_io_setup,
+      SYS_io_destroy,
+      SYS_remap_file_pages,
+      SYS_mbind,
+      SYS_set_mempolicy,
+      SYS_get_mempolicy,
+      SYS_migrate_pages,
+      SYS_move_pages,
+      SYS_mlock2,
+      SYS_pkey_mprotect}},
+    {"stat", 1, {SYS_stat}},
+    {"lstat", 1, {SYS_lstat}},
+    {"fstat", 3, {SYS_fstat, SYS_newfstatat, SYS_statx}},
+    {"stat_like",
+     5,
+     {SYS_stat, SYS_fstat, SYS_lstat, SYS_newfstatat, SYS_statx}},
+    {"statfs", 1, {SYS_statfs}},
+    {"fstatfs", 1, {SYS_fstatfs}},
+    {"statfs_like", 3, {SYS_ustat, SYS_statfs, SYS_fstatfs}},
+    {"pure",
+     8,
+     {SYS_getpid,
+      SYS_getuid,
+      SYS_getgid,
+      SYS_geteuid,
+      SYS_getegid,
+      SYS_getppid,
+      SYS_getpgrp,
+      SYS_gettid}},
+    {"syscall_never_fails",
+     12,
+     {SYS_getpid,
+      SYS_umask,
+      SYS_getuid,
+      SYS_getgid,
+      SYS_geteuid,
+      SYS_getegid,
+      SYS_getppid,
+      SYS_getpgrp,
+      SYS_setfsuid,
+      SYS_setfsgid,
+      SYS_personality,
+      SYS_gettid}},
+    {"max_args", 0, {}},
+    {"memory_mapping_change",
+     11,
+     {SYS_mmap,
+      SYS_mprotect,
+      SYS_munmap,
+      SYS_brk,
+      SYS_mremap,
+      SYS_shmat,
+      SYS_execve,
+      SYS_shmdt,
+      SYS_remap_file_pages,
+      SYS_execveat,
+      SYS_pkey_mprotect}},
```

Comment:
We don't support SYS_shmat,SYS_shmdt, SYS_remap_file_pages, SYS_pkey_mprotect.
Should we have entries for unsupported syscalls?

Reply:
I think it would be easier to keep this list forward thinking and in complete sync with the list in strace than go through all the groups again each time we add a syscall - what do you think?